mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala

```diff
@@ -85,31 +84,36 @@ class CheckCmd(name: String = "check", description: String = "Check a TLA+ speci
             inferpoly = Some(true)
         ),
     )
+
+    cfg.common.inputfile.foreach { file =>
+      if (newCfg.checker.config.isEmpty) {
+        // The older versions of apalache were loading a TLC config file of the same basename as the spec.
+        // We have flipped this behavior in version 0.25.0.
+        // Hence, warn the user that their config is not loaded by default.
+        val stem = FilenameUtils.removeExtension(file.getName())
+        val defaultConfig = new File(stem + ".cfg")
+        if (defaultConfig.exists()) {
+          val msg =
+            s"  > TLC config file found in specification directory. To enable it, pass --config=$defaultConfig."
+          logger.info(msg)
+        }
+      }
+    }
+
+    newCfg
   }
 
   def run() = {
-    // TODO: rm once OptionProvider is wired in
-    val cfg = configuration.left.map(err => new ConfigurationError(err)).toTry.get
+    val cfg = configuration.get
+    val options = OptionGroup.WithCheckerPreds(cfg).get
```

Comment:
Currently we're just letting the error handling here fall back to the existing exception handlers, but since these both return `Try`'s, and the methods end with an `Either`, we should be able to move to ADT based error handling for the run functions. I opened an issue for followup: https://github.com/informalsystems/apalache/issues/2106